testutil/exec.go

```diff
@@ -37,12 +37,18 @@ type MockCmd struct {
 	logFile string
 }
 
-var scriptTpl = `#!/bin/sh
-echo "$(basename "$0")" >> %[1]q
+// generates \0 to separate args
+// generates \00 to separate commands
+var scriptTpl = `#!/bin/bash
+echo -n "$(basename "$0")" >> %[1]q
+echo -n -e "\0" >> %[1]q
+
 for arg in "$@"; do
-    echo "$arg" >> %[1]q
+     echo -n "$arg" >> %[1]q
+     echo -n -e "\0"  >> %[1]q
 done
-echo >> %[1]q
+
+echo -n -e "\0" >> %[1]q
```

Comment:
I'm totally lost about what these changes do, I guess it's related to the MockAlso changes below. Can you briefly explain (a comment here, not in the code is fine). Thanks!